tests/testthat/test-try_stack.R

```diff
@@ -25,6 +25,16 @@ describe("with_retries", {
       "not enough retries"
     )
   })
-
-
 })
+
+describe("try_with_exit_code", {
+    test_that("it evaluates expr successfully", {
+    expect_true(try_with_exit_code(sum(1, 2)))
+  })
+    test_that("Expr evaluation fails and it exits R with specific exit code", {
+    with_mock(
+      `q` = function(status, ...) stop(status),
```

Comment:
Yes that does work. I suppose if it's only a machine using this (for the exit code), then we can remove `warning`, but perhaps for debugging stuff later, maybe use `print` or something else would make sense, too?